core/beneficiary/saver.go

```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
 	set     settler
 	ad      addressProvider
 	chainID int64
+	beneficiaryChangeKeeper
 }
 
-func newL1Saver(chainID int64, ad addressProvider, bc multiChainBC, set settler) *l1Saver {
+func newL1Saver(chainID int64, ad addressProvider, set settler, st storage) *l1Saver {
 	return &l1Saver{
-		chainID: chainID,
-		set:     set,
-		ad:      ad,
+		chainID:                 chainID,
+		set:                     set,
+		ad:                      ad,
+		beneficiaryChangeKeeper: beneficiaryChangeKeeper{st: st, chainID: chainID},
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
+
+const bucketBeneficiaryChangeStatus = "beneficiary-change-status"
+const (
+	// PENDING transaction is pending
+	PENDING SettleWithBeneficiaryState = "pending"
+	// COMPLETED transaction is completed
+	COMPLETED SettleWithBeneficiaryState = "completed"
+)
+
+type beneficiaryChangeKeeper struct {
+	st      storage
+	chainID int64
+}
+
+// BeneficiaryChangeStatus holds Beneficiary settlement transaction information
+type BeneficiaryChangeStatus struct {
+	State SettleWithBeneficiaryState
+	Error string
+}
+
+func (bcs *beneficiaryChangeKeeper) updateBeneficiaryChangeStatus(id identity.Identity, sbs SettleWithBeneficiaryState, settleError error) {
+	var errorMSG string
+	if settleError != nil {
+		errorMSG = settleError.Error()
```

Comment:
Im gonna be annoying here and say that it should be `errorMsg` if we want to be super idiomatic about things 🤷🏻